extensions/replication/ReplicationConfigValidator.js

```diff
@@ -73,6 +73,7 @@ const joiSchema = joi.object({
         minMPUSizeMB: joi.number().greater(0).default(20),
         probeServer: probeServerJoi.default(),
         circuitBreaker: joi.object().optional(),
+        sourceCheckIfSizeGreaterThanMB: joi.number().positive().default(100),
```

Comment:
May be a good occasion to review this option:
- can we find a better name (maxObjectSize ?)
- this option is weird, as it breaks crr semantics: what is the purpose of this option? Is it really used? Is there a matching option in cloudserver?

Reply:
This option simply adds a double check before changeing the status to COMPLETED, if an object is so large it takes time to replicate to avoid sending multiple times the same object for replication

Reply:
> this option is weird, as it breaks crr semantics: what is the purpose of this option? Is it really used? Is there a matching option in cloudserver?

The logic in S3C serves to avoid replicating already replicated objects. We already have the check in Zenko's replication code.
According to [this comment ](https://scality.atlassian.net/browse/S3C-3603?focusedCommentId=270438), the performance overhead of checking the replication state before replicating is not worth it, as in S3C and contrary to Zenko we do not create a new version in this case but only overwrite the previously replicated one. The check was only added for large objects as they consumed lots of resources and slowed things down when getting replicated.